src/Plugin/Block/AutologoutWarningBlock.php

```diff
@@ -8,33 +8,96 @@
+use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
  * Provides an 'Automated Logout info' block.
  *
  * @Block(
  *   id = "autologout_warning_block",
- *   admin_label = @Translation("Automated Logout info")
+ *   admin_label = @Translation("Automated Logout info"),
+ *   category = @Translation("User"),
  * )
  */
-class AutologoutWarningBlock extends BlockBase {
+class AutologoutWarningBlock extends BlockBase implements ContainerFactoryPluginInterface {
 
+  /**
+   * The module manager service.
+   *
+   * @var \Drupal\Core\Extension\ModuleHandlerInterface
+   */
+  protected $moduleHandler;
+
+  /**
+   * The date formatter service.
+   *
+   * @var \Drupal\Core\Datetime\DateFormatterInterface
+   */
+  protected $dateFormatter;
+
+  /**
+   * The config object for 'autologout.settings'.
+   *
+   * @var \Drupal\Core\Config\Config
+   */
+  protected $autoLogoutSettings;
+
+  /**
+   * Constructs a \Drupal\autologout\Plugin\Block\AutologoutWarningBlock object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $module_handler
+   *    The module manager service.
+   * @param \Drupal\Core\Datetime\DateFormatterInterface $date_formatter
+   *    The date formatter service.
+   * @param \Drupal\Core\Config\Config $autologout_settings
+   *    The config object for 'autologout.settings'.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ModuleHandlerInterface $module_handler, DateFormatterInterface $date_formatter, Config $autologout_settings) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->moduleHandler = $module_handler;
+    $this->dateFormatter = $date_formatter;
+    $this->autoLogoutSettings = $autologout_settings;
+  }
 
   /**
    * {@inheritdoc}
    */
-  public function defaultConfiguration() {
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('module_handler'),
+      $container->get('date.formatter'),
+      $container->get('config.factory')->get('autologout.settings')
+    );
+  }
 
-    $return = array();
-    if (\Drupal::moduleHandler()->moduleExists('jstimer')) {
-      if (!\Drupal::moduleHandler()->moduleExists('jst_timer')) {
-        drupal_set_message($this->t('The "Widget: timer" module must also be enabled for the dynamic countdown to work in the automated logout block.'), 'error');
-      }
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    // @todo: This is not the place where we should be doing this.
```

Comment:
This no way fits here. I was not aware of context of this message nor I saw this appearing anywhere. So I have commented it out. We will discuss the context and then put it at appropriate place.